bootkit/main.go

```diff
@@ -145,74 +149,51 @@ func main() {
 	fmt.Println(resp.ID)
 }
 
-func parsecmdline(cmdlines []string) (cfg tinkConfig, err error) {
-
+// parsecmdline will parse the command line.
+func parsecmdline(cmdlines []string) (cfg tinkConfig) {
 	for i := range cmdlines {
 		cmdline := strings.Split(cmdlines[i], "=")
-		if len(cmdline) != 0 {
-
-			// Find Registry configuration
-			if cmdline[0] == "docker_registry" {
-				cfg.registry = cmdline[1]
-			}
-			if cmdline[0] == "registry_username" {
-				cfg.username = cmdline[1]
-			}
-			if cmdline[0] == "registry_password" {
-				cfg.password = cmdline[1]
-			}
-
-			// Find Tinkerbell servers settings
-			if cmdline[0] == "packet_base_url" {
-				cfg.baseURL = cmdline[1]
-			}
-			if cmdline[0] == "tinkerbell" {
-				cfg.tinkerbell = cmdline[1]
-			}
-
-			// Find GRPC configuration
-			if cmdline[0] == "grpc_authority" {
-				cfg.grpcAuthority = cmdline[1]
-			}
-			if cmdline[0] == "grpc_cert_url" {
-				cfg.grpcCertURL = cmdline[1]
-			}
-
-			// Find the worker configuration
-			if cmdline[0] == "worker_id" {
-				cfg.workerID = cmdline[1]
-			}
+		if len(cmdline) == 0 {
+			continue
 		}
-	}
-	return
-}
 
-// DownloadFile will download a url to a local file. It's efficient because it will
-// write as it downloads and not load the whole file into memory.
-func DownloadFile(filepath string, url string) error {
+		// Find Registry configuration
+		if cmdline[0] == "docker_registry" {
```

Comment:
this should really use a switch statement: I'm surprised that the linting tool didn't pick this up :( 

```
switch cmd := cmdLine[0]; cmd {
  case "registry_username":
      ...
}
```

Reply:
Fixed